src/aligned_memory.rs

```diff
@@ -121,6 +121,11 @@ impl<const ALIGN: usize, T: AsRef<[u8]>> From<T> for AlignedMemory<ALIGN> {
     }
 }
 
+/// Returns true if `data` is aligned to `align`.
+pub fn is_memory_aligned(data: &[u8], align: usize) -> bool {
+    data.as_ptr().align_offset(align) == 0
```

Comment:
Nit: we've had some feedback that `align_offset` isn't the best choice:

> Description:
> The align_offset method is used to check whether a host memory address is aligned. However, that method is allowed to always return usize::MAX, even though
> the copy of the Rust standard library we have only does that if it is run at compile time.

> Recommendation:
> A better way to check for alignment is to simply check the remainder of the division of the address by the alignment.

Reply:
Ooh nice catch